### ISSUE TITLE
Print addditional information

### ISSUE THREAD

**Issue by dawidmalina** [keilerkonzept/terraform-module-versions#14]
Print addditional information:
- module source
- module path

The reason is to quickly Identify which exactly module require update as we can have (like in my case) local modules which includes other remote modules.

@sgreben, @jack17529 what do you think about such change?

**Comment by dawidmalina** [comment]
Simple example why it's important for me. This is example of my terraform code structure
```
examples
├── emea
│      └── main.tf
├── apac
│      └── main.tf
└── noram
       └── main.tf
```

Modules within main.tf are basically identical but might have different modules version and currently it's difficult to identify which exactly need to be upgraded.

**Comment by sgreben** [comment]
I agree on the necessity - the path is in fact currently being printed in the JSON output. I left it out of the markdown output because then the lines tend to end up being excessively long.. perhaps we should add some formatting options to the -pretty mode?

**Comment by jack17529** [comment]
Yes, the code already does have it. The output would be something like -
`"consul_aws" in examples/0.12.x.tf`

**Comment by sgreben** [comment]
Hmm. How about using a shortest-unique-suffix of the path in the `-pretty` output? So in the above example from @dawidmalina we'd get `emea/main.tf`, `apac/main.tf`, `noram/main.tf`

**Comment by jack17529** [comment]
@sgreben I would be more happy with the full file path. How about another table with just 3 fields `Path`,`Latest Matching` and `Latest Overall`.

**Comment by dawidmalina** [comment]
I think that json and markup output should be fully equal in the content and the only difference should be about way to present those data.

Btw we could remove some code duplication if the engine will always generates json data and last step will determine if the console output should be pure json or markup based on the json data.

**Comment by sgreben** [comment]
> I think that json and markup output should be fully equal in the content and the only difference should be about way to present those data.

I disagree on this - they have different use-cases (machine-readable vs. human-readable). At least by default, `-pretty` should be optimized for being easily readable, whereas JSON should contain _all_ the info (e.g. for piping into `jq`). 

I am open to making the `-pretty` view customizable though. I need to think about how to do this in a UX-good way still.

**Comment by dawidmalina** [comment]
In this case I would say we are abusing what the `pretty` means. Usually (take a `jq` as an example) `pretty` are used to make the content displayed in human readable way but it's *does not* modify the content - it's just a different way of presenting the data.

It's fully up to you :) and I will respect your decision. Can you also take under consideration different type of outputs like junit for example?

**Comment by dawidmalina** [comment]
One more thought also related to https://github.com/keilerkonzept/terraform-module-versions/issues/12

What could be implemented in the cli:

| Arguments | Actions |
| --- | --- |
| ` ` | List all configured modules as json |
| `-p` or alias | List all configured modules as markup |
| `-u` or alias | List all modules which require updates as json (here we could cover #12 by default) |
| `-u -p` or aliases | List all modules which require updates as markup (here we could cover #12 by default) |
| `-r` or `--report` | Additional arg which will define whether extra information should be added into json or markup output |

On the other hand we could rethink cli and define sub-commands and steering arguments. For example:
- `terraform-module-versions version` - will do the same as now
- `terraform-module-versions list` - will be used to print all configured modules with their current version, name, sources, etc
- `terraform-module-versions check` - will be used to check if for configured modules we can find new versions
- `terraform-module-versions update` - maybe some day we could have auto update :)

Then we could use extra args like:
- `-o json` (default), `-o markup`, `-o junit` - to define output format
- `-q` - to suppress log output
- `-e` - to exit with exit-code when update found

**Comment by sgreben** [comment]
Agreed! I also think subcommand style is the way to go, especially as we're adding flags that interact. I'm traveling right now, but Sat/Sun I'll have a bit more time to think about the design. Also junit output is a good idea, it can be considered a kind of linter ("out-of-date dependencies") from a CI standpoint after all :)

**Comment by dawidmalina** [comment]
If we are talking about junit here is code example: https://github.com/terraform-linters/tflint/pull/843/files

**Comment by sgreben** [comment]
addressed in #25, including subcommands & junit output
